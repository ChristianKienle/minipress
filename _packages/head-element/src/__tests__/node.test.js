// @ts-check

const Node = require('./../node')

describe('Node', () => {
  it('renders simple link (empty, no attributes)', () => {
    expect(new Node('link', {}, Node.IS_EMPTY, '').toString()).toMatchSnapshot()
  })

  it('renders simple link (not empty, no attributes)', () => {
    expect(new Node('link', {}, Node.IS_NOT_EMPTY, '').toString()).toMatchSnapshot()
  })

  it('renders simple link (not emptry with single attr)', () => {
    expect(new Node('link', { test: 'hihi' }, Node.IS_NOT_EMPTY, '').toString()).toMatchSnapshot()
  })

  it('renders simple link (emptry with single attr)', () => {
    expect(new Node('link', { test: 'hihi' }, Node.IS_EMPTY, '').toString()).toMatchSnapshot()
  })
})
