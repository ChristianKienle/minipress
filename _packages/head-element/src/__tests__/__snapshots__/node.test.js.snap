// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node renders simple link (emptry with single attr) 1`] = `"<link test=\\"hihi\\">"`;

exports[`Node renders simple link (empty, no attributes) 1`] = `"<link>"`;

exports[`Node renders simple link (not emptry with single attr) 1`] = `"<link test=\\"hihi\\"></link>"`;

exports[`Node renders simple link (not empty, no attributes) 1`] = `"<link></link>"`;
